tests/testthat/test-utils-format.R

```diff
@@ -0,0 +1,18 @@
+context("utils-format")
+
+test_that("pluralise works correctly", {
+  expect_identical(pluralise("[an] index(es)", c("x")), "an index")
+  expect_identical(pluralise("[an] index(es)", c("x", "y")), "indexes")
+})
+
+test_that("pluralise leaves alone parentheses / square brackets that have spaces inside", {
+  expect_identical(pluralise("[an] index(es )", c("x")), "an index(es )")
+  expect_identical(pluralise("[an ] index(es)", c("x", "y")), "[an ] indexes")
+})
```

Comment:
`pluralise("[an ] index(es )", c("x"))` returns `"an  index(es )"`. There's 1 object, so we singularize, i.e. 1) strip `[`, and 2) remove `()` and its content. However, since `()` contains space in this case, the `()` and its content are left alone.

My understanding is that you want to preserve parens / square brackets if they are part of the error message. However, even in that case there wouldn't be any space inside, e.g. `Invalid column (please be careful!)`. Hence I'm a little confused about why we preserve parens / brackets with spaces inside.